presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -13,114 +13,172 @@
  */
 package com.facebook.presto;
 
+import com.facebook.presto.execution.QueryManagerConfig;
+import com.facebook.presto.execution.TaskManagerConfig;
+import com.facebook.presto.memory.MemoryManagerConfig;
+import com.facebook.presto.spi.session.SessionPropertyMetadata;
+import com.facebook.presto.sql.analyzer.FeaturesConfig;
+import com.google.common.collect.ImmutableList;
 import io.airlift.units.DataSize;
 
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static com.facebook.presto.spi.session.SessionPropertyMetadata.booleanSessionProperty;
+import static com.facebook.presto.spi.session.SessionPropertyMetadata.integerSessionProperty;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+
 public final class SystemSessionProperties
 {
     private static final String OPTIMIZE_HASH_GENERATION = "optimize_hash_generation";
-    private static final String DISTRIBUTED_JOIN = "distributed_join";
-    private static final String HASH_PARTITION_COUNT = "hash_partition_count";
+    public static final String DISTRIBUTED_JOIN = "distributed_join";
```

Comment:
Having mixed private/public is confusing. Probably best to make them all public since they are extremely public values anyway.